src/shims/unix/linux/fd/epoll.rs

```diff
@@ -0,0 +1,44 @@
+use crate::*;
+
+use crate::shims::unix::fs::FileDescriptor;
+
+use rustc_data_structures::fx::FxHashMap;
+use std::io;
+
+#[derive(Clone, Debug, Default)]
+/// An `Epoll` connects file handles and epoll events
+pub struct Epoll {
+    pub file_descriptors: FxHashMap<i32, EpollEvent>,
+}
+
+#[derive(Clone, Debug)]
+/// Epoll Events associate events with data
```

Comment:
Well I saw that much from the fields. ;) But what *are* 'events' and 'data'? I guess 'events' are things like "bytes available for read" or "buffer space available for write", but data...?

Reply:
We don't really know yet. They aren't used yet, only initialized. Once there's a test actually needing them, we'll figure it out

Reply:
Uh... epoll is documented somewhere, right? You didn't just come up with them?
Then please add a reference to whatever docs they are come from, and also add a note saying that Miri currently doesn't do anything with them.